app/experimenter/nimbus-ui/src/lib/test-utils.tsx

```diff
@@ -76,24 +82,32 @@ export const assertSerializerMessages = async (
   for (const [field, errors] of Object.entries(messages)) {
     // An object of serializer messages contains values of either:
     // - an array of strings
-    // - an array of objects containing this same thing, indicating nested fields
-    // We start by checking if the field contains nested fields
-    if (typeof errors[0] === "object") {
-      let index = 0;
-      for (const set of errors) {
-        for (const [innerField, innerErrors] of Object.entries(set)) {
-          await screen.findByText(innerErrors.join(", "), {
-            selector: `[data-for="${snakeToCamelCase(
-              field,
-            )}[${index}].${snakeToCamelCase(innerField)}"]`,
-          });
+    // - an object containing this same thing
```

Comment:
Oh man, this is getting dicey lol. I don't have a suggestion for improvement though, I think we definitely need the individual errors on the branches the way you did it for the UI.

Reply:
Yeah I am not a fan of the variety of shapes a serializer response can be